src/test/java/iudx/file/server/apiserver/integrationtests/RestAssuredConfiguration.java

```diff
@@ -0,0 +1,92 @@
+package iudx.file.server.apiserver.integrationtests;
+
+import io.restassured.RestAssured;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import iudx.file.server.authenticator.TokenSetup;
+import iudx.file.server.configuration.Configuration;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.restassured.RestAssured.*;
+import static iudx.file.server.authenticator.TokensForITs.*;
+
+/**
+ * JUnit5 extension to allow {@link RestAssured} configuration to be injected into all integration
+ * tests using {@link org.junit.jupiter.api.extension.ExtendWith}.
+ */
+public class RestAssuredConfiguration implements BeforeAllCallback {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestAssuredConfiguration.class);
+
+    @Override
+    public void beforeAll(ExtensionContext context) {
+        Vertx vertx = Vertx.vertx();
+        Configuration fileServerConfig = new Configuration();
+        JsonObject config = fileServerConfig.configLoader(0, vertx);
+        String testHost = config.getString("ip");
+        //  String testHost = System.getProperty("intTestHost");
+        //System.out.println("testHost:"+testHost);
+        JsonObject config2 = fileServerConfig.configLoader(1, vertx);
+        String authServerHost = config2.getString("authHost");
+
+        if (testHost != null) {
+            baseURI = "http://" + testHost;
+        } else {
+            baseURI = "http://localhost";
+        }
+
+        String testPort = config.getString("httpPort");
+        // String testPort = System.getProperty("intTestPort");
+        if (testPort != null) {
+            port = Integer.parseInt(testPort);
+        } else {
+            port = 8443;
+        }
```

Comment:
Need to get the testOnDepl environment variable for integration tests on deployment

Reply:
Added this part in RestAssuredConfiguration file:

 boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }

Reply:
boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }


Added this part in RestAssuredConfiguration file.

Reply:
boolean testOnDepl = Boolean.parseBoolean(System.getProperty("intTestDepl"));
        if (testOnDepl) {
            String testHost = "file-test.iudx.io";
            baseURI = "https://" + testHost;
            port = 443;
        } else {
            String testHost = System.getProperty("intTestHost");

            if (testHost != null) {
                baseURI = "http://" + testHost;
            } else {
                baseURI = "http://localhost";
            }

            String testPort = System.getProperty("intTestPort");

            if (testPort != null) {
                port = Integer.parseInt(testPort);
            } else {
                port = 8081;
            }
        }